Src/ResistanceOperationOverhaul/Classes/UILadderSquadUpgradeScreen.uc

```diff
@@ -728,21 +743,9 @@ simulated function SetSoldierGear()
 
 	mc.QueueString("Secondary Weapon");//secondary
 	
-	//we need to handle the reaper claymore
-	ItemTemplateManager = class'X2ItemTemplateManager'.static.GetItemTemplateManager();
-	SoldierClassTemplate = class'X2SoldierClassTemplateManager'.static.GetSoldierClassTemplateManager().FindSoldierClassTemplate(Soldier.GetSoldierClassTemplateName());
-	if (SoldierClassTemplate.DataName == 'Reaper')
-	{
-		EquipmentTemplate = X2EquipmentTemplate(ItemTemplateManager.FindItemTemplate('Reaper_Claymore'));
-		mc.QueueString(EquipmentTemplate.strImage);
-		mc.QueueString(EquipmentTemplate.GetItemFriendlyName());
-	}
-	else
-	{
-		equippedItem = Soldier.GetItemInSlot(eInvSlot_SecondaryWeapon, NewGameState, false);
-		mc.QueueString(equippedItem.GetMyTemplate().strImage);
-		mc.QueueString(equippedItem.GetMyTemplate().GetItemFriendlyNameNoStats());
-	}
+	equippedItem = Soldier.GetItemInSlot(eInvSlot_SecondaryWeapon, NewGameState, false);
```

Comment:
Tested again, It works fine. Displays the claymore correctly, and switches to a new weapon Icon (Which didn't before when it was changed) when switched away from a claymore. I made it this way cause with Proficiency Reapers, It would always display a claymore despite it being a pistol.